src/sage/rings/function_field/maps.py

```diff
@@ -45,1081 +35,39 @@
-        True
-        sage: e = L.random_element()
-        sage: h(h(e,1),1) == 2*h(e,2)
-        True
-        sage: h(h(h(e,1),1),1) == 3*2*h(e,3)
-        True
-    """
-    def __init__(self, field):
-        """
-        Initialize.
-
-        TESTS::
-
-            sage: K.<x> = FunctionField(QQ); _.<Y> = K[]
-            sage: L.<y> = K.extension(Y^3 + x + x^3*Y)
-            sage: h = L.higher_derivation()
-            sage: TestSuite(h).run(skip=['_test_category'])
-        """
-        FunctionFieldHigherDerivation.__init__(self, field)
-
-        self._separating_element = field(field.base_field().gen())
-
-        # cache computed higher derivatives to speed up later computations
-        self._cache = {}
-
-    def _call_with_args(self, f, args, kwds):
-        """
-        Call the derivation with ``args`` and ``kwds``.
-
-        EXAMPLES::
-
-            sage: K.<x> = FunctionField(QQ); _.<Y> = K[]
-            sage: L.<y> = K.extension(Y^3 + x + x^3*Y)
-            sage: h = L.higher_derivation()
-            sage: e = L.random_element()
-            sage: h(h(e,1),1) == 2*h(e,2)  # indirect doctest
-            True
-        """
-        return self._derive(f, *args, **kwds)
-
-    def _derive(self, f, i, separating_element=None):
-        """
-        Return ``i``-th derivative of ``f`` with respect to the separating
-        element.
-
-        EXAMPLES::
-
-            sage: K.<x> = FunctionField(QQ); _.<Y> = K[]
-            sage: L.<y> = K.extension(Y^3 + x + x^3*Y)
-            sage: h = L.higher_derivation()
-            sage: y^3
-            -x^3*y - x
-            sage: h._derive(y^3,0)
-            -x^3*y - x
-            sage: h._derive(y^3,1)
-            (-21/4*x^4/(x^7 + 27/4))*y^2 + ((-9/2*x^9 - 45/2*x^2)/(x^7 + 27/4))*y + (-9/2*x^7 - 27/4)/(x^7 + 27/4)
-        """
-        F = self._field
-
-        if separating_element is None:
-            x = self._separating_element
-            xderinv = 1
-        else:
-            x = separating_element
-            xderinv = ~(x.derivative())
-
-        try:
-            cache = self._cache[separating_element]
-        except KeyError:
-            cache = self._cache[separating_element] = {}
-
-        if i == 0:
-            return f
-
-        try:
-            return cache[f, i]
-        except KeyError:
-            pass
-
-        s = i
-        e = f
-        while s > 0:
-            e = xderinv * e.derivative() / F(s)
-            s -= 1
-
-        der = e
-
-        cache[f, i] = der
-        return der
+lazy_import("sage.rings.function_field.derivations", (
+    "FunctionFieldDerivation",
+    "FunctionFieldDerivation_rational",
+    "FunctionFieldDerivation_separable",
+    "FunctionFieldDerivation_inseparable",
+    "FunctionFieldHigherDerivation",
+    "RationalFunctionFieldHigherDerivation_global",
+    "FunctionFieldHigherDerivation_global",
+    "FunctionFieldHigherDerivation_char_zero",
+), deprecation=35230)
```

Comment:
I don't think that these deprecations are necessary. They are internal modules, and not intended to be used directly by users.

Reply:
OK, done.